conf/u-boot.conf

```diff
@@ -3,17 +3,19 @@
 ## @useflag uboot_extra_env Set this to extra env in u-boot. Will be appended
 ##          to the existing default environment. Is empty by default.
 ## @useflag uboot_config_file Filename of the uboot config to use.
+## @useflag uboot_image makefile target to build.
 ## @useflag uboot_maxsize Enables checking of sizecheck. Default disabled.
 ##          Set this to the maximum size allowed of your u-boot image.
-CLASS_FLAGS += "uboot_config uboot_config_file uboot_extra_env uboot_maxsize"
+CLASS_FLAGS += "uboot_config uboot_config_file uboot_extra_env uboot_maxsize uboot_image"
 
 DEFAULT_USE_uboot_config ?= "${MACHINE}_config"
 DEFAULT_USE_uboot_extra_env ?= ""
+DEFAULT_USE_uboot_image ?= "u-boot.bin"
 
 COMPATIBLE_MACHINES = ".*"
 
 ## @var UBOOT_IMAGE Name of the image to build. Is u-boot.bin by default.
-UBOOT_IMAGE ?= "u-boot.bin"
+UBOOT_IMAGE ?= "${USE_uboot_image}"
```

Comment:
Could you add
```
DEFAULT_USE_uboot_image = "u-boot.bin"
```
Which should be a sane default, and give us backwards compatibility.